server/handler/query.go

```diff
@@ -0,0 +1,88 @@
+package handler
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/src-d/gitbase-playground/server/serializer"
+)
+
+type queryRequest struct {
+	Query string `json:"query"`
+	Limit int    `json:"limit"`
+}
+
+// genericVals returns a slice of interface{}, each one a pointer to a NullString
+func genericVals(nColumns int) []interface{} {
+	columnVals := make([]sql.NullString, nColumns)
+	columnValsPtr := make([]interface{}, nColumns)
+
+	for i := range columnVals {
+		columnValsPtr[i] = &columnVals[i]
+	}
+
+	return columnValsPtr
+}
+
+// Query returns a function that forwards an SQL query to gitbase and returns
+// the rows as JSON
+func Query(db *sql.DB) RequestProcessFunc {
+	return func(r *http.Request) (*serializer.Response, error) {
+		var queryRequest queryRequest
+		body, err := ioutil.ReadAll(r.Body)
+		defer r.Body.Close()
+		if err == nil {
+			err = json.Unmarshal(body, &queryRequest)
+		}
+
+		if err != nil {
+			return nil, err
+		}
+
+		// TODO (carlosms) this only works if the query does not end in ;
+		// and does not have a limit. It will also fail for queries like
+		// DESCRIBE TABLE
```

Comment:
gitbase doesn't support anything besides SELECT, SHOW, DESCRIBE.
Add limit makes sense only to SELECT queries. Ending `;` also very easy to fix:
```
query = strings.TrimRight(strings.TrimSpace(query), ";")
```
not perfect but better than nothing

Reply:
:+1: Implemented the todo with your suggestion in e6deab2